### PR TITLE
Add mouse support for the menu in edit and paint

### LINKED ISSUE #416: Mouse support for paint and edit menus
Currently, in order to access the menu bar of the `edit` and `paint` programs, you must press <kbd>Ctrl</kbd>. This is perfectly fine, but it's rather curious that you cannot use the mouse to either open or interact with the menu.

I think there's three bits of functionality we need here:
 - Clicking the menu bar will open it.
 - Clicking out of the menu bar will close it.
 - Clicking a button will close the menu bar, and act upon that action.

This is so obvious, I'm kinda mad I didn't do it years ago.

### PR DESCRIPTION
**Pull request by JakobDev**
Closes #416

### CODE REVIEW

**Comment by lupus590** [comment]
I case you have not noticed, you might want to fix the linter errors.

**Comment by SquidDev** [comment]
I haven't done any thorough testing yet, but fiddling around on mobile, this looks really good. Thank you!

File: src/main/resources/assets/computercraft/lua/rom/programs/edit.lua [hunk 1]
```diff
@@ -61,7 +61,7 @@ if peripheral.find("printer") then
 end
 table.insert(tMenuItems, "Exit")
 
-local sStatus = "Press Ctrl to access menu"
+local sStatus = "Press Ctrl or click here to access menu"
```
Comment (SquidDev) [inline src/main/resources/assets/computercraft/lua/rom/programs/edit.lua:64]: On normal computers, this should probably still show "Press Ctrl to access menu" - we don't have mouse support there!

File: src/main/resources/assets/computercraft/lua/rom/programs/fun/advanced/paint.lua [hunk 2]
```diff
@@ -261,27 +261,58 @@ local function accessMenu()
...
             if selection == k then
                 term.setTextColour(colours.yellow)
                 local ox = term.getCursorPos()
                 term.write("[" .. string.rep(" ", #v) .. "]")
                 term.setCursorPos(ox + 1, h)
+                nStartPos = ox + 1
                 term.setTextColour(colours.white)
                 term.write(v)
-                term.setCursorPos(term.getCursorPos() + 1, h)
+                nEndPos = term.getCursorPos()
```
Comment (SquidDev) [inline src/main/resources/assets/computercraft/lua/rom/programs/fun/advanced/paint.lua:304]: Is this missing a `+ 1`? We currently have this, which is a little broken:

Slightly broken menu

That said, do we need to keep track of the positions? You don't appear to within edit, which does end up being a little shorter/cleaner.

File: src/main/resources/assets/computercraft/lua/rom/programs/edit.lua (hunk 1 above)
Comment (Lemmmy) [reply src/main/resources/assets/computercraft/lua/rom/programs/edit.lua:64]: Double check how the message looks on advanced pocket computers, might be worth trying to shorten it

E.g. changing 'access' to 'open'

Comment (SquidDev) [reply src/main/resources/assets/computercraft/lua/rom/programs/edit.lua:64]: Pocket computers have a shorter message anyway:

https://github.com/SquidDev-CC/CC-Tweaked/blob/da419b24e701229eee3afa06315969c5d2a4f972/src/main/resources/assets/computercraft/lua/rom/programs/edit.lua#L65-L67

Comment (Lemmmy) [reply src/main/resources/assets/computercraft/lua/rom/programs/edit.lua:64]: Perfect

**Pull request merged**